Felienne/hedy
# Implements support for RTL languages

**Description**
Currently we only support a LTR (left-to-right) layout. However, multiple languages such as Arabic and Hebrew are using a  RTL (right-to-left) structure. In this PR we implement a dynamic front-end setting to support both layouts. We set a global value to either `ltr` or `rtl` on load, based on the selected language. Based on this value we set the `dir` attribute on the body of `layout.html` to the correct value. This value is stored in a `const dir` within `app.ts` to enable us to, for example, set the ace editor direction to the right one.

With the merging of #1683 we are using Tailwind v3, with built-in LTR and RTL support. However, their support makes the styling of our front-end a bit complexer. For example, instead of using `pl-6` to create a padding of `6` on the left we would have to use `ltr:pl-6 rtl:pr-6` to support both layouts. As a padding to the left on LTR would mean a padding to the right on RTL. Every time we use either `margin`, `padding` or just `right/left` in our styling we have to define this for both LTR and RTL.This is not an ideal situation and styling will cost a bit more time and debugging. But, this is the only way Tailwind currently support LTR and RTL layouts.

**Fix for**
This PR fixes #1679.

**How to test**
Set the `g.dir` value in `app.py` on line 280 fixed to `rtl` and notice that the complete website changes. Navigate through all pages and verify that all margins and paddings are still correct. As this is only a front-end task everything should still work fine.

**Known bugs**

- [x] ~~Shadow on commands window does not work with RTL (stays on rights side of window)~~
- [x] ~~Re-sizing main editor through developers toggle with RTL does not align lines correctly~~
- [x] ~~Programming experience checkboxes on edit-profile are aligned on the wrong side of the page~~
